cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -81,28 +82,31 @@ def __init__(
         self._event_loop = asyncio.get_event_loop()
         self._query_semaphore = asyncio.Semaphore(max_concurrent_requests)
 
-    def batch_prompt(
+    def _batch_prompt(
         self,
-        prompts: List[str],
-        options: Union[None, TLMOptions, List[Union[TLMOptions, None]]] = None,
+        prompts: Collection[str],
+        options: Union[None, TLMOptions, Collection[Union[TLMOptions, None]]] = None,
         timeout: Optional[float] = None,
-        retries: int = 0,
+        retries: int = 1,
     ) -> List[TLMResponse]:
         """Run batch of TLM prompts.
 
         Args:
-            prompts (List[str]): list of prompts to run
-            options (None | TLMOptions | List[TLMOptions  |  None], optional): list of options (or instance of options) to pass to prompt method. Defaults to None.
+            prompts (Collection[str]): list or other iterable of prompts to run
+            options (None | TLMOptions | List[TLMOptions  |  None], optional): collection of options (or instance of options) to pass to prompt method. Defaults to None.
             timeout (Optional[float], optional): timeout (in seconds) to run all prompts. Defaults to None.
-            retries (int): number of retries to attempt for each individual prompt. Defaults to 0.
+            retries (int): number of retries to attempt for each individual prompt in case of error. Defaults to 1.
 
         Returns:
             List[TLMResponse]: TLM responses for each prompt (in supplied order)
         """
-        if not isinstance(options, list):
-            options = [options for _ in prompts]
+        if is_collection(options):
+            options_collection = cast(Collection[Union[TLMOptions, None]], options)
+        else:
```

Comment:
```python3
def is_collection(obj: object) -> bool:
    return hasattr(obj, "__iter__") and hasattr(obj, "__len__")

options = {"metadata": "true", "logprobs": "10", "best_of": "3"}
is_collection(options)
```

This returns True. Above string errors if `_batch_prompt()` is called with a single options dictionary and a list of prompts (not sure if the desire behavior here is to project the single options dictionary into an options_collection